tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -0,0 +1,171 @@
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithDefaultPlatformLoggingEnabledFalse()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithDefaultPlatformLoggingEnabled(false).Build();
+            Assert.IsFalse(pca.AppConfig.IsDefaultPlatformLoggingEnabled);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithWithEnablePiiLoggingTrue()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithEnablePiiLogging(true).Build();
+            Assert.IsTrue(pca.AppConfig.EnablePiiLogging);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithWithEnablePiiLoggingFalse()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithEnablePiiLogging(false).Build();
+            Assert.IsFalse(pca.AppConfig.EnablePiiLogging);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithHttpClientFactory()
+        {
+            var httpClientFactory = new MyHttpClientFactory();  
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithHttpClientFactory(httpClientFactory).Build();
+            Assert.AreEqual(httpClientFactory, pca.AppConfig.HttpClientFactory);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithLoggingCallback()
+        {
+            var pca = PublicClientApplicationBuilder
+                      .Create(MsalTestConstants.ClientId).WithLoggingCallback(((level, message, pii) => { })).Build();
+
+            Assert.IsNotNull(pca.AppConfig.LoggingCallback);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithLoggingLevel()
+        {
+            var pca = PublicClientApplicationBuilder
+                      .Create(MsalTestConstants.ClientId).WithLoggingLevel(LogLevel.Verbose).Build();
+
+            Assert.AreEqual(LogLevel.Verbose, pca.AppConfig.LogLevel);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithRedirectUri()
+        {
+            var pca = PublicClientApplicationBuilder
+                      .Create(MsalTestConstants.ClientId).WithRedirectUri("http://some_redirect_uri/").Build();
+
+            Assert.AreEqual("http://some_redirect_uri/", pca.AppConfig.RedirectUri);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithTenantId()
+        {
+            var pca = PublicClientApplicationBuilder
+                      .Create(MsalTestConstants.ClientId).WithTenantId("a_tenant id").Build();
+
+            Assert.AreEqual("a_tenant id", pca.AppConfig.TenantId);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithTelemetryCallback()
+        {
+            var pca = PublicClientApplicationBuilder
+                      .Create(MsalTestConstants.ClientId).WithTelemetryCallback((events => { })).Build();
```

Comment:
>(events => { }) [](start = 80, length = 15)

would this in essence be the TelemetryCallback? Would it make sense to evalaute that those are the same (by specifying the callback using an object allowing you to validate the object references?